test/common/http/utility_test.cc

```diff
@@ -124,6 +124,18 @@ TEST(HttpUtility, TestParseCookie) {
   EXPECT_EQ(value, "abc123");
 }
 
+TEST(HttpUtility, TestParseCookieBadValues) {
+  TestHeaderMapImpl headers{{"cookie", "token1=abc123; = "},
+                            {"cookie", "token2=abc123;   "},
+                            {"cookie", "; token3=abc123;"},
+                            {"cookie", "=; token4=\"abc123\""}};
+
+  EXPECT_EQ(Utility::parseCookieValue(headers, "token1"), "abc123");
```

Comment:
The error handling is the break/continue case. It is excercised if `=` is missing from a part of the cookie. This is handled in several of the cookie headers given in this test IIRC. I'm trying to run locally with and without `source/*` changes to verify.